core/jobs/batch_jobs/audit_rule_validation_jobs_test.py

```diff
@@ -0,0 +1,1047 @@
+# coding: utf-8
+#
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS-IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for jobs.batch_jobs.exp_validation_jobs."""
+
+from __future__ import annotations
+
+import datetime
+
+from core import feconf
+from core.constants import constants
+from core.domain import state_domain
+from core.jobs import job_test_utils
+from core.jobs.batch_jobs import audit_rule_validation_jobs
+from core.jobs.types import job_run_result
+from core.platform import models
+
+(exp_models, opportunity_models) = models.Registry.import_models(
+    [models.NAMES.exploration, models.NAMES.opportunity])
+
+
+class ExpAuditRuleChecksJobTest(job_test_utils.JobTestBase):
+
+    JOB_CLASS = (
+        audit_rule_validation_jobs.ExpAuditRuleChecksJob
+    )
+
+    EXPLORATION_ID_1 = '1'
+    EXPLORATION_ID_2 = '2'
+    EXPLORATION_ID_3 = '3'
+    EXPLORATION_ID_4 = '4'
+    EXPLORATION_ID_5 = '5'
+
+    # DragAndDrop Interaction.
+    EXP_1_STATE_1 = state_domain.State.create_default_state(
```

Comment:
Please add `# type: ignore[no-untyped-call]` statement here. Because these functions are defined in state_domain.py and state_domain.py is not typed yet
Thanks!

Reply:
Thanks